source/dotnet/Test/AdaptiveCards.Test/SerializationTests.cs

```diff
@@ -78,9 +78,19 @@ public void TestSkippingUnknownElements()
             var result = AdaptiveCard.FromJson(json);
 
             Assert.IsNotNull(result.Card);
-            Assert.AreEqual(1, result.Card.Body.Count);
-            Assert.AreEqual(1, result.Card.Actions.Count);
+            Assert.AreEqual(2, result.Card.Body.Count);
+            Assert.AreEqual(2, result.Card.Actions.Count);
             Assert.AreEqual(2, result.Warnings.Count);
+
+            // check first unknown element
+            var unknown_elem = (AdaptiveUnknownElement)result.Card.Body[0];
+            Assert.AreEqual(unknown_elem.Type, "IDunno");
+            Assert.AreEqual(unknown_elem.AdditionalProperties["text"], "Hello");
```

Comment:
Nit: I used to find adding messages in the assertion checks very useful when looking at logs to get more context debugging -- not something that may be useful? Just a consideration

Reply:
Worth taking a look at in the future, I think. Our tests need a bit of love overall :)

---
In reply to: [272374244](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r272374244) [](ancestors = 272374244)